testing/load-test/v2/common.js

```diff
@@ -0,0 +1,229 @@
+import {check} from 'k6';
+import http from 'k6/http';
+import {Counter, Trend} from 'k6/metrics';
+
+import {randomIntBetween} from "https://jslib.k6.io/k6-utils/1.0.0/index.js";
+
+export let randomByteString = function (size) {
+    let str = '';
+    for (let i = 0; i < size; i++) {
+        // First 128 chars (US-ASCII) take 1 byte to UTF-8 encode.
+        str += String.fromCharCode(randomIntBetween(0, 128));
+    }
+    return str;
+}
+
+const baseUrl = 'http://localhost:9391';
+
+let produceBinaryToTopicRequestLatency = new Trend('ProduceBinaryToTopicRequestLatency', true);
+let produceBinaryToTopicRequestCount = new Counter('ProduceBinaryToTopicRequestCount');
+let produceBinaryToTopicMessageCount = new Counter('ProduceBinaryToTopicMessageCount');
+let produceBinaryToTopicByteCount = new Counter('ProduceBinaryToTopicByteCount');
+export let produceBinaryToTopic = function (topicName, records) {
+    const url = `${baseUrl}/topics/${topicName}`;
+    const body = {records};
+    const headers = {
+        'Accept': 'application/vnd.kafka.v2+json',
+        'Content-Type': 'application/vnd.kafka.binary.v2+json'
+    };
+    const tags = {name: 'ProduceBinaryToTopic'};
+    let response = http.post(url, JSON.stringify(body), {headers, tags});
+    check(response, {'ProduceBinaryToTopic: Status is 200': response => response.status === 200});
+    produceBinaryToTopicRequestLatency.add(response.timings.duration);
+    produceBinaryToTopicRequestCount.add(1);
+    if (response.status === 200) {
+        produceBinaryToTopicMessageCount.add(records.length);
+        produceBinaryToTopicByteCount.add(
+            records.reduce((total, record) => total + record.key.length + record.value.length));
+    }
+    return response;
+}
+
+let produceJsonToTopicRequestLatency = new Trend('ProduceJsonToTopicRequestLatency', true);
+let produceJsonToTopicRequestCount = new Counter('ProduceJsonToTopicRequestCount');
+let produceJsonToTopicMessageCount = new Counter('ProduceJsonToTopicMessageCount');
+let produceJsonToTopicByteCount = new Counter('ProduceJsonToTopicByteCount');
+export let produceJsonToTopic = function (topicName, records) {
+    const url = `${baseUrl}/topics/${topicName}`;
+    const body = {records};
+    const headers = {
+        'Accept': 'application/vnd.kafka.v2+json',
+        'Content-Type': 'application/vnd.kafka.json.v2+json'
+    };
+    const tags = {name: 'ProduceJsonToTopic'};
+    let response = http.post(url, JSON.stringify(body), {headers, tags});
```

Comment:
possibly naive thoughts:

wondering various things - why isn't this async?, does k6 use worker threads?, what is the perf cost of various bits of this JS code?. generally, I'm wondering if there is a bottleneck and if that is k6 or REST Proxy? network? I guess i'm thinking i wouldn't be surprised to find REST Proxy more performant than what this test framework can do (and if it's not, why isn't it), so i'm wondering if you're implicitly testing the performance of the test framework to any degree. also, if you're testing over a network other than localhost, I'm guessing you may be able to saturate the network, unless it's a really fast one, so wondering what throughput you're seeing relative to network maximum.

given you seem to be holding VUs constant in the tests, I assume the idea is for latency in request responses to provide back pressure on throughput making this a meaningful thing to measure. you can expect throughput characteristics to change quite a bit with number of concurrent connections I think though, so i think you definitely want to be testing a range. i guess you have probably at least experimented with this on an ad-hoc basis. and it would of course make the tests more time consuming / burdensome to run.